### PR TITLE
Propagate marked attributes from primary constructor parameters to backing fields.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #73920.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Emit3/Semantics/PrimaryConstructorTests.cs [hunk 1]
```diff
@@ -22360,5 +22361,258 @@ public record {{typeKind}} C4
...
+public class Preserve1Attribute : Attribute { }
+
+[CompilerLoweringPreserve]
+[AttributeUsage(AttributeTargets.Parameter)]
+public class Preserve2Attribute : Attribute { }
+";
+
+            string source2 = @"
+public record " + keyword + @" Test1(
+    [Preserve1]
```
[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit3/Semantics/PrimaryConstructorTests.cs:22437]: #73920 explicitly asks  to not copy the attribute for records:
> Cases where attributes should not flow:
>
> - Primary constructor parameters of record types: the compiler allows adding [property: ...] annotations to primary constructor parameters of record types ...

[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit3/Semantics/PrimaryConstructorTests.cs:22437]: A user is able to explicitly apply attributes to backing fields in records

**Review by cston — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Review by 333fred — APPROVED** [review]

**Pull request merged**